sinks/sinks_service.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+// Adapted for Orb project, modifications licensed under MPL v. 2.0:
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package sinks
+
+import (
+	"context"
+	"fmt"
+	"github.com/mainflux/mainflux"
+	"github.com/ns1labs/orb/pkg/errors"
+	"github.com/ns1labs/orb/sinks/backend"
+)
+
+var (
+	ErrCreateSink                 = errors.New("failed to create Sink")
+	ErrConflictSink               = errors.New("entity already exists")
+	ErrUnsupportedContentTypeSink = errors.New("unsupported content type")
+)
+
+func (svc sinkService) CreateSink(ctx context.Context, token string, sink Sink) (Sink, error) {
+
+	mfOwnerID, err := svc.identify(token)
+	if err != nil {
+		return Sink{}, err
+	}
+
+	sink.MFOwnerID = mfOwnerID
+
+	if backend.HaveBackend(sink.Backend) {
+		err = backend.GetBackend(sink.Backend).Connect(sink.Config)
+		if err != nil {
+			sink.Status = "not connected"
+			sink.Error = fmt.Sprint(err)
+		} else {
+			sink.Status = "connected"
+			sink.Error = ""
+		}
+	}
```

Comment:
it needs to be an error if the back end does not exist. this is where we will also use the ValidateOnly flag - if that is true, it should return here before saving